tests/test_spatial_combine_transforms.py

```diff
@@ -0,0 +1,137 @@
+            (RandFlipd, {"prob": 0.9, "keys": "img"}),
+            (RandRotated, {"prob": 0.9, "range_x": np.pi / 4, "keys": "img"}),
+            (RandZoomd, {"prob": 0.9, "mode": "bilinear", "keep_size": False, "keys": "img", "align_corners": False}),
+        ],
+    ],
+]
+
+TEST_3D = [
+    [
+        (2, 64, 64, 64),
+        [
+            (Spacing, {"pixdim": (1.2, 1.5, 0.9), "padding_mode": "zeros", "dtype": torch.float32}),
+            (Orientation, {"axcodes": "RAS"}),
+            (Resize, {"spatial_size": (32, 32, 32), "mode": "nearest"}),
+            (RandFlip, {"prob": 0.9}),
+            (RandRotate, {"prob": 0.9, "range_x": np.pi / 4}),
+            (RandZoom, {"prob": 0.9, "mode": "nearest", "keep_size": False}),
+        ],
+    ],
+    [
+        (2, 64, 64, 64),
+        [
+            (Spacingd, {"pixdim": (1.2, 1.5, 0.9), "padding_mode": "zeros", "dtype": torch.float32, "keys": "img"}),
+            (Orientationd, {"axcodes": "RAS", "keys": "img"}),
+            (Resized, {"spatial_size": (32, 32, 32), "mode": "nearest", "keys": "img"}),
+            (RandFlipd, {"prob": 0.9, "keys": "img"}),
+            (RandRotated, {"prob": 0.9, "range_x": np.pi / 4, "keys": "img"}),
+            (RandZoomd, {"prob": 0.9, "mode": "nearest", "keep_size": False, "keys": "img"}),
+        ],
+    ],
+]
+
+
+class CombineLazyTest(unittest.TestCase):
+    @parameterized.expand(TEST_2D + TEST_3D)
+    def test_combine_transforms(self, input_shape, funcs):
+        for seed in [10, 100, 1000, 10000]:
+            _funcs = []
+            for _func, _params in funcs:
+                _funcs.append(_func(**_params))
+            is_map = isinstance(_funcs[0], MapTransform)
+            data = torch.randint(low=1, high=10, size=input_shape).float()
+            im = MetaTensor(data, meta={"a": "b", "affine": np.eye(len(input_shape))})
+            input_data = {"img": im} if is_map else im
+            # non lazy
+            non_lazy_result = input_data
+            for _func in _funcs:
+                if isinstance(_func, Randomizable):
+                    _func.set_random_state(seed=seed)
+                non_lazy_result = _func(non_lazy_result)
+            expected = non_lazy_result["img"] if is_map else non_lazy_result
+
+            # lazy
+            pending_result = input_data
+            for _func in _funcs:
+                _func.lazy_evaluation = True
+                if isinstance(_func, Randomizable):
+                    _func.set_random_state(seed=seed)
+                pending_result = _func(pending_result)
+            pending_result = pending_result["img"] if is_map else pending_result
+
+            assert_allclose(pending_result.peek_pending_affine(), expected.affine)
+            assert_allclose(pending_result.peek_pending_shape(), expected.shape[1:4])
+            # # TODO: how to test final result?
+            # init_param = funcs[-1][1]
```

Comment:
should we enable tests on final result? Should the results between lazy and non-lazy be close? On my local test, `atol=1.0` is not enough.

Reply:
code to compare:
```py
import matplotlib.pyplot as plt
from monai.visualize import matshow3d

a = plt.subplot(1,3,1); matshow3d(result.array[0], fig=a)
b = plt.subplot(1,3,2); matshow3d(expected.array[0], fig=b)
c = plt.subplot(1,3,3); matshow3d(np.abs(expected.array[0] - result.array[0]), fig=c)
plt.show()
```